pubnub/pnconfiguration.py

```diff
@@ -36,10 +35,10 @@ def __init__(self):
         self._heartbeat_interval = PNConfiguration.DEFAULT_HEARTBEAT_INTERVAL
 
     def validate(self):
-        assert self.uuid is None or isinstance(self.uuid, str)
+        PNConfiguration.validate_not_empty_string(self.uuid)
 
-        if not self.uuid:
-            self.uuid = utils.uuid()
+    def validate_not_empty_string(value):
+        assert value and isinstance(value, str), "UUID missing or invalid type"
```

Comment:
When I give uuid = " ", it is not throwing exception.
To handle empty spaces, we can refactor it by adding strip and check length after strip.
`assert value and isinstance(value, str) and len(value.strip()) > 0 , "UUID missing or invalid type"`